tensorflow/python/framework/ops.py

```diff
@@ -686,6 +686,15 @@ def __copy__(self):
   # with ndarrays.
   __array_priority__ = 100
 
+  def __array__(self):
+    raise NotImplementedError("Cannot convert a symbolic Tensor ({}) to a numpy"
+                              " array.".format(self.name))
+
+  def __len__(self):
+    raise AttributeError("len is not well defined for symbolic Tensors. ({}) "
```

Comment:
the current TypeError error is probably fine.  why do we need this?

Reply:
You're right; it should be a TypeError not an AttributeError. I'll fix it. The idea of this is twofold:
1) To more clearly surface which Tensor is causing the issue, since this can appear quite deep in library code.
2) To explain to users who might not be familiar with 1.x semantics (either because they've only used 2.0 or are coming from something like scikit-learn) why this thing that looks like a numpy array and clearly has a shape is refusing to tell them its length.

(Also, TIL that if you have text in a response and resolve a conversation github will just drop it...)